src/ansys/fluent/core/utils/file_transfer_service.py

```diff
@@ -232,15 +213,37 @@ def upload(
         FileNotFoundError
             If a file does not exist.
         """
-        files = _get_files(file_name, self.host_mount_path)
+        files = _get_files(file_name)
         if self.client:
             for file in files:
-                is_file_on_remote = self.file_exists_on_remote(os.path.basename(file))
-                if is_file_on_remote:
+                full_file_name, is_file_on_remote = self.file_exists_on_remote(
+                    os.path.basename(file)
+                )
+                if is_file_on_remote and full_file_name.samefile(file):
                     warnings.warn(
                         f"\n{file} with the same name exists at the remote location.\n",
                         UserWarning,
                     )
+                    return
+                elif (
+                    is_file_on_remote
+                    and os.path.isfile(file)
+                    and not full_file_name.samefile(file)
+                ):
+                    warnings.warn(
+                        f"\n{file} with the same name and different content exists at the remote location. "
+                        f"It's replaced with the given file.\n",
+                        UserWarning,
+                    )
+                    full_file_name.unlink()
+                    self.client.upload_file(
+                        local_filename=file,
+                        remote_filename=(
+                            remote_file_name
+                            if remote_file_name
+                            else os.path.basename(file)
+                        ),
+                    )
```

Comment:
If local and remote files are with same name but with different content then upload the local file.

Reply:
I think we should use this  logic in download as well.